feat: Integrate Morph vaults and underlying assets

sgh94 opened the pull request:
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama):


##### Twitter Link:


##### List of audit links if any:


##### Website Link:


##### Logo (High resolution, will be shown with rounded borders):


##### Current TVL:


##### Treasury Addresses (if the protocol has treasury)


##### Chain:


##### Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)


##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap, leave empty if not listed): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)


##### Short Description (to be shown on DefiLlama):


##### Token address and ticker if any:


##### Category (full list at https://defillama.com/categories) *Please choose only one:


##### Oracle Provider(s): Specify the oracle(s) used (e.g., Chainlink, Band, API3, TWAP, etc.):
##### Implementation Details: Briefly describe how the oracle is integrated into your project:
##### Documentation/Proof: Provide links to documentation or any other resources that verify the oracle's usage:

##### forkedFrom (Does your project originate from another project):


##### methodology (what is being counted as tvl, how is tvl being calculated):


##### Github org/user (Optional, if your code is open source, we can track activity):

llamabutler commented:
The adapter at projects/mitosis exports TVL: 
        
 
 
    linea                     21.17 M
    ethereum                  12.40 M
    arbitrum                  5.44 M
    scroll                    4.34 M
    bsc                       1.58 M
    blast                     1.03 M
    mode                      931.63 k
    mantle                    631.00 k
    optimism                  245.09 k
    manta                     57.29 k
    morph                     0
    
    total                    47.82 M 
    
    RPC request count per chain
    ┌──────────┬────────┐
    │ (index)  │ Values │
    ├──────────┼────────┤
    │ ethereum │ 5      │
    │ scroll   │ 4      │
    │ arbitrum │ 5      │
    │ optimism │ 4      │
    │ mode     │ 4      │
    │ manta    │ 1      │
    │ blast    │ 4      │
    │ linea    │ 5      │
    │ bsc      │ 3      │
    │ mantle   │ 3      │
    │ morph    │ 1      │
    └──────────┴────────┘

llamabutler commented:
The adapter at projects/mitosis exports TVL: 
        
 
 
    linea                     21.03 M
    ethereum                  12.39 M
    arbitrum                  5.46 M
    scroll                    4.34 M
    bsc                       1.57 M
    blast                     1.03 M
    mode                      932.28 k
    mantle                    634.74 k
    optimism                  241.29 k
    morph                     115.08 k
    manta                     57.33 k
    
    total                    47.79 M 
    
    RPC request count per chain
    ┌──────────┬────────┐
    │ (index)  │ Values │
    ├──────────┼────────┤
    │ ethereum │ 5      │
    │ scroll   │ 4      │
    │ arbitrum │ 5      │
    │ optimism │ 4      │
    │ mode     │ 4      │
    │ manta    │ 1      │
    │ blast    │ 4      │
    │ linea    │ 5      │
    │ bsc      │ 3      │
    │ mantle   │ 3      │
    │ morph    │ 1      │
    └──────────┴────────┘

realdealshaman commented:
Hey @sgh94 I've added usdc to the morph core assets list so that it can be referenced correctly. Please revert the last change

llamabutler commented:
The adapter at projects/mitosis exports TVL: 
        
 
 
    linea                     21.32 M
    ethereum                  12.55 M
    arbitrum                  5.50 M
    scroll                    4.39 M
    bsc                       1.57 M
    blast                     1.05 M
    mode                      944.72 k
    mantle                    624.18 k
    optimism                  239.07 k
    morph                     198.67 k
    manta                     58.10 k
    
    total                    48.45 M 
    
    RPC request count per chain
    ┌──────────┬────────┐
    │ (index)  │ Values │
    ├──────────┼────────┤
    │ ethereum │ 5      │
    │ scroll   │ 4      │
    │ arbitrum │ 5      │
    │ optimism │ 4      │
    │ mode     │ 4      │
    │ manta    │ 1      │
    │ blast    │ 4      │
    │ linea    │ 5      │
    │ bsc      │ 3      │
    │ mantle   │ 3      │
    │ morph    │ 1      │
    └──────────┴────────┘

sgh94 commented:
> Hey @sgh94 I've added usdc to the morph core assets list so that it can be referenced correctly. Please revert the last change

Thanks, I’ve reverted it.
